wee_slack.py

```diff
@@ -1898,13 +1898,23 @@ class SlackUser(object):
     """
 
     def __init__(self, **kwargs):
-        # We require these two things for a vaid object,
+        # We require these two things for a valid object,
         # the rest we can just learn from slack
         self.identifier = kwargs["id"]
-        self.slack_name = kwargs["name"]
-        self.name = kwargs["name"]
+        self.profile = {}  # in case it's not in kwargs
         for key, value in kwargs.items():
             setattr(self, key, value)
+
+        if self.profile.get("display_name"):
+            self.slack_name = self.profile["display_name"]
+            # TODO: configuration options here for 'what character to replace with'
+            # and 'whether to use normalized display name or real one'.
+            self.name = self.profile["display_name"].replace(' ', '')
+        else:
+            # No display name set. This should never happen, but if it does,
```

Comment:
That this should never happens isn't correct. [The docs](https://api.slack.com/methods/users.info) say:
> The profile hash contains as much information as the user has supplied in the default profile fields: first_name, last_name, real_name, display_name, skype, and the image_* fields. Only the image_* fields are guaranteed to be included. Data that has not been supplied may not be present at all, may be null or may contain the empty string ("").

On my testing team, multiple users has `display_name` set to the empty string. (I see that this is the same cause as my previous comment, which is now fixed, but you should change this comment as well).

Reply:
Yeah, forgot to update the comment along with the code. Done now.